### DISCUSSION TITLE
Buggy Samsung Workaround on Oppo

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#3073] [Bug reports]
Could you please add Oppo to the buggy Samsung workaround?
I have installed the xDrip Version where the workaround is enabled for every Phone and it's the only version working stable with a Dexcom G6 and Android 13, because of the background restrictions. 

Without the workaround, I'm missing ever second reading from my G6

**Comment by Navid200** [comment]
Please provide a link to manufacturer and model.

**Reply by Navid200** [reply]
Also, provide a link to the xDrip version that does not have the problem you are experiencing.

Please also show your settings.

**Reply by Miquest** [reply]
https://www.oppo.com/chde/smartphones/series-find-x/find-x5-pro/

Manufacturer is Oppo and the model ist the Find X5 Pro.

Here is the Link to my Phone model, but all OnePlus and Oppo devices suffer with that Problem if they've installed Android 12 or later. 

I have installed Version 2123032817 of xDrip Plus and the Settings are configured as follows:

![Screenshot_2023-09-09-16-31-08-48](https://github.com/NightscoutFoundation/xDrip/assets/91971937/7afefcbb-ca3e-4dc8-abca-baedcd74eb97)

**Reply by Navid200** [reply]
You have disabled minimize scanning.  
That's incorrect for Android 10.

And you have enabled scan for G5 constantly.  Again, something you should not.

So, we are not on common ground.  There are so many factors that could cause problems with proper operation of xDrip.
These recommendations are meant to help:  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

So, just because the problem goes away when you use that version of xDrip is not enough for us to change xDrip.
Please install the latest Nightly release: https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.09.08
Please follow the recommended settings.

If the problem persists, enable detail OB1 debugging: https://navid200.github.io/xDrip/docs/Dexcom/OB1Debug.html
After enabling the detail logs, wait for the readings to occur only once every 10 minutes.   After it happens a few times, go ahead and upload as explained on that page.  And post here with the note to identify the logs are from you related to this thread.

**Reply by Miquest** [reply]
We had many Threads in the past with exactly the same problem. 
So why do I have to verify the existence of this problem again?

There ist exactly one version working, the one version with unified Buggy Samsung workaround. So, I can patch the .apk myself to show you that IT IS a problem with xDrip itself If you want me to do so. 

Sorry If I'm Not as Kind as I should be, but I first reported this Problem about one year ago and since then, Nothing happened to fix that problem.

**Comment by Navid200** [comment]
Oppo has been added.

**Reply by Navid200** [reply]
https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.10.17

**Reply by Miquest** [reply]
Thank you so much for realizing that!
